core/domain/voiceover_domain.py

```diff
@@ -0,0 +1,219 @@
+        self.entity_id = entity_id
+        self.entity_type = entity_type
+        self.entity_version = entity_version
+        self.language_accent_code = language_accent_code
+        self.voiceovers = voiceovers
+
+    def to_dict(self) -> EntityVoiceoverDict:
+        """Returns the dict representation of the EntityVoiceover object.
+
+        Returns:
+            EntityVoiceoverDict. The dict representation of the
+            EntityVoiceover object.
+        """
+        voiceovers_dict = {}
+        for content_id, voiceover_type_to_voiceover in self.voiceovers.items():
+            voiceovers_dict[content_id] = {
+                'manual': voiceover_type_to_voiceover['manual'].to_dict(),
+                'auto': voiceover_type_to_voiceover['auto'].to_dict()
+            }
+        return {
+            'entity_id': self.entity_id,
+            'entity_type': self.entity_type,
+            'entity_version': self.entity_version,
+            'language_accent_code': self.language_accent_code,
+            'voiceovers': voiceovers_dict
+        }
+
+    @classmethod
+    def from_dict(
+        cls, entity_voiceover_dict: EntityVoiceoverDict
+    ) -> EntityVoiceover:
+        """Creates the EntityVoiceover from the given dict.
+
+        Args:
+            entity_voiceover_dict: EntityVoiceoverDict. The dict
+                representation of the EntityVoiceover object.
+
+        Returns:
+            EntityVoiceover. The EntityVoiceover object created using the
+            given dict.
+        """
+        content_id_to_voiceovers = {}
+        for content_id, voiceover_type_to_voiceover_dict in (
+                entity_voiceover_dict['voiceovers'].items()):
+            content_id_to_voiceovers[content_id] = {
+                'manual': state_domain.Voiceover.from_dict(
+                    voiceover_type_to_voiceover_dict['manual']),
+                'auto': state_domain.Voiceover.from_dict(
+                    voiceover_type_to_voiceover_dict['auto'])
+            }
+
+        return cls(
+            entity_voiceover_dict['entity_id'],
+            entity_voiceover_dict['entity_type'],
+            entity_voiceover_dict['entity_version'],
+            entity_voiceover_dict['language_accent_code'],
+            content_id_to_voiceovers
+        )
+
+    def validate(self) -> None:
+        """Validates the EntityVoiceover object."""
+        if not isinstance(self.entity_type, str):
+            raise utils.ValidationError(
+                'entity_type must be a string, recieved %r' % self.entity_type)
+        if not isinstance(self.entity_id, str):
+            raise utils.ValidationError(
+                'entity_id must be a string, recieved %r' % self.entity_id)
+        if not isinstance(self.entity_version, int):
+            raise utils.ValidationError(
+                'entity_version must be an int, recieved %r' %
+                self.entity_version)
+        if not isinstance(self.language_accent_code, str):
```

Comment:
Should we do more validation on these -- would they follow a specific type of string format, say?